### PR TITLE
Support ember 3.10, drops support for decorators

### PR DESCRIPTION
**Pull request by GavinJoyce**
extracted from https://github.com/GavinJoyce/ember-cp-validations/pull/1
fixes https://github.com/offirgolan/ember-cp-validations/issues/640 and resolves some of the issues described in https://github.com/offirgolan/ember-cp-validations/issues/635

This PR is a breaking change as it drops support for decorators which have proven somewhat unstable. They were introduced a year ago in https://github.com/offirgolan/ember-cp-validations/pull/583 so they are likely used by some apps. A good path forward would be to create `ember-cp-validations-decorators` which can depend on this addon and provide decorators support.

### CODE REVIEW

**Comment by GavinJoyce** [comment]
It seems that's I'm hitting some `ember-decorators` failing tests:

```
 Error: Assertion Failed: computed decorators must return an instance of an Ember ComputedProperty descriptor, received function COMPUTED_DECORATOR(target
```

**Comment by GavinJoyce** [comment]
Unsurprisingly, the decorators issue is causing some versions to fail:

Screen Shot 2019-05-28 at 15 47 46

File: tests/dummy/app/models/company.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-  @attr('string') name;
-}
+// export const Validations = buildValidations({
+//   name: validator('presence', { presence: true, description: 'Name' })
+// });
+
+// @hasValidations(Validations)
+// export default class Company extends Model {
+//   @attr('string') name;
+// }
```
Comment (GavinJoyce) [inline tests/dummy/app/models/company.js]: Uncommenting this results in the following failing test:

```
Build Error (broccoli-persistent-filter:Babel > [Babel: ember-cp-validations]) in dummy/models/company.js

/Users/gavinjoyce/dev/opensource/ember-cp-validations/dummy/models/company.js: Support for the experimental syntax 'decorators-legacy' isn't currently enabled (13:1):

  11 | });
  12 |
> 13 | @hasValidations(Validations)
     | ^
  14 | export default class Company extends Model {
  15 |   @attr('string') name;
  16 | }
```

**Comment by GavinJoyce** [comment]
This is passing on all scenarios except for beta and canary:

Screen Shot 2019-05-29 at 10 27 31

I've opened an issue to track that: https://github.com/offirgolan/ember-cp-validations/issues/642

**Comment by GavinJoyce** [comment]
We're running this in production for @intercom at the moment

**Comment by cah-danmonroe** [comment]
`GavinJoyce/ember-cp-validations.git#8ba9fec36a2b3269bb95b66b0a2c68a832d6921d` Fixes over 70 errors we had going from 3.to 3.10.  Please merge

File: addon/-private/options.js [hunk 2]
```diff
@@ -18,11 +17,6 @@ export default class Options {
     const optionKeys = keys(options);
     const createParams = { [OPTION_KEYS]: optionKeys, model, attribute };
 
-    // If any of the options is a CP, we need to create a custom class for it
-    if (optionKeys.some(key => isDescriptor(options[key]))) {
```
Comment (offirgolan) [inline addon/-private/options.js]: Is this change necessary if the `isDescriptor` logic is working?

File: tests/dummy/app/models/company.js [hunk 3]
```diff
@@ -1,16 +1,10 @@
...
-  buildValidations
-} from 'ember-cp-validations';
+import DS from 'ember-data';
+import { validator, buildValidations } from 'ember-cp-validations';
 
 export const Validations = buildValidations({
   name: validator('presence', { presence: true, description: 'Name' })
 });
 
-@hasValidations(Validations)
```
Comment (offirgolan) [inline tests/dummy/app/models/company.js]: Whats the need to drop decorators support?

**Comment by offirgolan** [comment]
@GavinJoyce thanks for taking the time to resolve all of the current issues! Once this PR gets merged, I'll release a new beta version. 

If the decorator support has been causing so many issues, I say we remove it altogether and I'll make a note of it in the new beta version's changelog. Feel free to remove it completely if thats necessary to get this working in 3.10+.

File: addon/-private/options.js (hunk 2 above)
Comment (GavinJoyce) [reply addon/-private/options.js]: Good point, thanks. I'll add it back

**Comment by GavinJoyce** [comment]
@offirgolan I've removed the remaining references to decorators, this should be good to go now

**Comment by offirgolan** [comment]
Thanks @GavinJoyce! Released under `v4.0.0-beta.9`.

**Comment by Alonski** [comment]
Awesome work @GavinJoyce !!!

**Comment by jderr-mx** [comment]
Awesome work, thank you!

**Comment by alejandrodevs** [comment]
Awesome. Thank you all!

**Comment by BryanCrotaz** [comment]
I'm seeing this bug with beta 9 and ember 3.10.2
